src/dmd/backend/cv8.d

```diff
@@ -639,73 +684,78 @@ void cv8_outsym(Symbol *s)
 
     uint sr;
     uint base;
-    switch (s->Sclass)
+    switch (s.Sclass)
     {
         case SCparameter:
         case SCregpar:
         case SCshadowreg:
-            if (s->Sfl == FLreg)
+            if (s.Sfl == FLreg)
             {
-                s->Sfl = FLpara;
+                s.Sfl = FLpara;
                 cv8_outsym(s);
-                s->Sfl = FLreg;
+                s.Sfl = FLreg;
                 goto case_register;
             }
-            base = Para.size - BPoff;    // cancel out add of BPoff
+            base = cast(uint)(Para.size - BPoff);    // cancel out add of BPoff
             goto L1;
 
         case SCauto:
-            if (s->Sfl == FLreg)
+            if (s.Sfl == FLreg)
                 goto case_register;
         case_auto:
-            base = Auto.size;
+            base = cast(uint)Auto.size;
         L1:
-#if 1
+static if (1)
+{
             // Register relative addressing
-            buf->reserve(2 + 2 + 4 + 4 + 2 + len + 1);
-            buf->writeWordn( 2 + 4 + 4 + 2 + len + 1);
-            buf->writeWordn(0x1111);
-            buf->write32(s->Soffset + base + BPoff);
-            buf->write32(typidx);
-            buf->writeWordn(I64 ? 334 : 22);       // relative to RBP/EBP
+            buf.reserve(cast(uint)(2 + 2 + 4 + 4 + 2 + len + 1));
+            buf.writeWordn(cast(uint)(2 + 4 + 4 + 2 + len + 1));
+            buf.writeWordn(0x1111);
+            buf.write32(cast(uint)(s.Soffset + base + BPoff));
+            buf.write32(typidx);
+            buf.writeWordn(I64 ? 334 : 22);       // relative to RBP/EBP
             cv8_writename(buf, id, len);
-            buf->writeByte(0);
-#else
+            buf.writeByte(0);
+}
+else
+{
             // This is supposed to work, implicit BP relative addressing, but it does not
-            buf->reserve(2 + 2 + 4 + 4 + len + 1);
-            buf->writeWordn( 2 + 4 + 4 + len + 1);
-            buf->writeWordn(S_BPREL_V3);
-            buf->write32(s->Soffset + base + BPoff);
-            buf->write32(typidx);
+            buf.reserve(2 + 2 + 4 + 4 + len + 1);
+            buf.writeWordn( 2 + 4 + 4 + len + 1);
+            buf.writeWordn(S_BPREL_V3);
+            buf.write32(s.Soffset + base + BPoff);
+            buf.write32(typidx);
             cv8_writename(buf, id, len);
-            buf->writeByte(0);
-#endif
+            buf.writeByte(0);
+}
             break;
 
         case SCbprel:
             base = -BPoff;
             goto L1;
 
         case SCfastpar:
-            if (s->Sfl != FLreg)
-            {   base = Fast.size;
+            if (s.Sfl != FLreg)
+            {   base = cast(uint)Fast.size;
                 goto L1;
             }
             goto L2;
 
         case SCregister:
-            if (s->Sfl != FLreg)
+            if (s.Sfl != FLreg)
                 goto case_auto;
+            goto L2;
```

Comment:
nit: `goto case;` would be a closer translation.

Reply:
ok